File: packages/ruby/model/robot.rb
class Robot
    attr_reader :id
    attr_reader :player_id
    attr_reader :is_teammate
    attr_reader :x
    attr_reader :y
    attr_reader :z
    attr_reader :velocity_x
    attr_reader :velocity_y
    attr_reader :velocity_z
    attr_reader :radius
    attr_reader :nitro_amount
    attr_reader :touch
    attr_reader :touch_normal_x
    attr_reader :touch_normal_y
    attr_reader :touch_normal_z
    def initialize(json)
        @id = json["id"]
        @player_id = json["player_id"]
        @is_teammate = json["is_teammate"]
        @x = json["x"]
        @y = json["y"]
        @z = json["z"]
        @velocity_x = json["velocity_x"]
        @velocity_y = json["velocity_y"]
        @velocity_z = json["velocity_z"]
        @radius = json["radius"]
        @nitro_amount = json["nitro_amount"]
        @touch = json["touch"]
        @touch_normal_x = json["touch_normal_x"]
        @touch_normal_y = json["touch_normal_y"]
        @touch_normal_z = json["touch_normal_z"]
    end
end

File: packages/javascript/model/Robot.js
/**
 * Created by Quake on 18.12.2018
 */
'use strict';

function Robot() {

}

Robot.prototype.read = function(json) {
    if (typeof json == 'string') {
        json = JSON.parse(json);
    }

    this.id = json.id;
    this.player_id = json.player_id;
    this.is_teammate = json.is_teammate;
    this.x = json.x;
    this.y = json.y;
    this.z = json.z;
    this.velocity_x = json.velocity_x;
    this.velocity_y = json.velocity_y;
    this.velocity_z = json.velocity_z;
    this.radius = json.radius;
    this.nitro_amount = json.nitro_amount;
    this.touch = json.touch;
    if (this.touch) {
        this.touch_normal_x = json.touch_normal_x;
        this.touch_normal_y = json.touch_normal_y;
        this.touch_normal_z = json.touch_normal_z;
    }
}

module.exports = Robot;